staging/src/k8s.io/apiserver/pkg/storage/cacher.go

```diff
@@ -369,6 +369,12 @@ func (c *Cacher) Get(ctx context.Context, key string, resourceVersion string, ob
 		return err
 	}
 
+	if !c.ready.check() && getRV == 0 {
```

Comment:
We are always contending on the lock.  You can't check the check without holding the lock